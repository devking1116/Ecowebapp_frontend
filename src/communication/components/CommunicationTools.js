/* @flow */
import React from 'react';
import { Link } from 'react-router';
import reactCSS from 'reactcss';
import { Menu, Icon, Select } from 'semantic-ui-react';

import { FillRemaining } from '../../core';

type Props = {
  status: 'active' | 'in-active' | 'none',
  tab: 'details' | 'comments' | 'activity',
};

/**
 * A component that displays the Communication
 */
export default function CommunicationTools({ status, tab, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      wrapper: {
        position: 'relative',
      },
      tabMenu: {
        paddingLeft: 10,
      },
    },
  });

  return (
    <div style={styles.wrapper} {...otherProps}>
      <Menu secondary>
        <Menu.Item>
            <Icon name="commenting" />
            <span>Comment</span>
        </Menu.Item>
        <Menu.Item>
          <Icon name="tag" />
          <span>Logbook</span>
        </Menu.Item>
        <Menu.Item>
          <Icon name="share" />
          <span>Share</span>
        </Menu.Item>
        <Menu.Item>
          <Icon name="edit" />
          <span>Edit</span>
        </Menu.Item>
        <Menu.Item>
          <Icon name="trash outline" />
          <span>Delete</span>
        </Menu.Item>
      </Menu>
      <Menu tabular style={styles.tabMenu}>
        <Menu.Item active={tab === 'details'}>
          <Link to="/details">Details</Link>
        </Menu.Item>
        <Menu.Item active={tab === 'comments'}>
          <Link to="/comments">Comments</Link>
        </Menu.Item>
        <Menu.Item active={tab === 'activity'}>
          <Link to="/activity">Activity</Link>
        </Menu.Item>
      </Menu>
    </div>
  );
}
