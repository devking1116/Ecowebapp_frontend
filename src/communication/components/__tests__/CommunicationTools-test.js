import React from 'react';
import renderer from 'react-test-renderer';

import CommunicationTools from '../CommunicationTools';

test('it displays the given tab as selected.', () => {
  const componentDetails = renderer.create(
    <CommunicationTools tab="details" />
  );
  const componentComments = renderer.create(
    <CommunicationTools tab="comments" />
  );
  const componentActivity = renderer.create(
    <CommunicationTools tab="activity" />
  );
  const treeDetails = componentDetails.toJSON();
  const treeComments = componentComments.toJSON();
  const treeActivity = componentActivity.toJSON();

  expect(treeDetails).toMatchSnapshot();
  expect(treeComments).toMatchSnapshot();
  expect(treeActivity).toMatchSnapshot();
});
