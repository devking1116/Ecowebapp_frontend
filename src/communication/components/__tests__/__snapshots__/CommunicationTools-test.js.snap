exports[`test it displays the given tab as selected. 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="ui secondary menu">
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="commenting icon" />
      <span>
        Comment
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="tag icon" />
      <span>
        Logbook
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="share icon" />
      <span>
        Share
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="edit icon" />
      <span>
        Edit
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="trash outline icon" />
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    className="ui tabular menu"
    style={
      Object {
        "paddingLeft": 10,
      }
    }>
    <div
      className="active item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Details
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Comments
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Activity
      </a>
    </div>
  </div>
</div>
`;

exports[`test it displays the given tab as selected. 2`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="ui secondary menu">
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="commenting icon" />
      <span>
        Comment
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="tag icon" />
      <span>
        Logbook
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="share icon" />
      <span>
        Share
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="edit icon" />
      <span>
        Edit
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="trash outline icon" />
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    className="ui tabular menu"
    style={
      Object {
        "paddingLeft": 10,
      }
    }>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Details
      </a>
    </div>
    <div
      className="active item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Comments
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Activity
      </a>
    </div>
  </div>
</div>
`;

exports[`test it displays the given tab as selected. 3`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="ui secondary menu">
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="commenting icon" />
      <span>
        Comment
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="tag icon" />
      <span>
        Logbook
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="share icon" />
      <span>
        Share
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="edit icon" />
      <span>
        Edit
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <i
        aria-hidden="true"
        className="trash outline icon" />
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    className="ui tabular menu"
    style={
      Object {
        "paddingLeft": 10,
      }
    }>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Details
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Comments
      </a>
    </div>
    <div
      className="active item"
      onClick={[Function]}>
      <a
        onClick={[Function]}
        style={Object {}}>
        Activity
      </a>
    </div>
  </div>
</div>
`;
